codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -186,35 +184,31 @@ void onUnknownFrame0(ChannelHandlerContext ctx, byte frameType, int streamId, Ht
         @Override
         public int onDataRead(final ChannelHandlerContext ctx, int streamId, ByteBuf data,
                 int padding, boolean endOfStream) throws Http2Exception {
-            // Check if we received a data frame for a stream which is half-closed
             Http2Stream stream = connection.requireStream(streamId);
 
-            verifyGoAwayNotReceived();
+            if (stream.isResetSent() || streamCreatedAfterGoAwaySent(stream)) {
+                int bytesToReturn = data.readableBytes() + padding;
```

Comment:
This is also duplicated later on in this method.  Consider defining this 1 time and use it in both places.  I think it will only go unused in the error path, and it is relatively small overhead in that condition anyways.